protocols/gossipsub/src/behaviour.rs

```diff
@@ -592,14 +592,17 @@ where
         // Transform the data before building a raw_message.
         let transformed_data = self
             .data_transform
-            .outbound_transform(&topic, data.clone())?;
+            .outbound_transform(&topic.clone(), data.clone())?;
+
+        let max_transmit_size_for_topic = self.config.max_transmit_size_for_topic(&topic);
 
         // check that the size doesn't exceed the max transmission size.
-        if transformed_data.len() > self.config.max_transmit_size() {
+        if transformed_data.len() > max_transmit_size_for_topic {
             return Err(PublishError::MessageTooLarge);
         }
 
-        let raw_message = self.build_raw_message(topic, transformed_data)?;
+        let raw_message = self.build_raw_message(topic.clone(), transformed_data)?;
+        let mesh_n = self.config.mesh_n_for_topic(&topic);
```

Comment:
very small nit, 

If you move this one line up, you can avoid the clone on `topic`